### DISCUSSION TITLE
Why is ipywidgets installed via micropip rather than preinstalled?

### DISCUSSION THREAD

**Discussion by dsblank** [jupyterlite/jupyterlite#236] [Q&A] [answered]
In the examples, I see:

```python
import micropip
await micropip.install("ipywidgets")
```

Why is `ipywidgets` not already in collection of preinstalled Python libraries, like `matplotlib`?

**Comment by jtpio** [comment, marked as answer]
Thanks @dsblank for raising this.

`matplotlib` is preinstalled in `pyodide`: https://github.com/pyodide/pyodide/tree/main/packages/matplotlib

And `ipywidgets` is not. We considered adding it by default in https://github.com/jupyterlite/jupyterlite/pull/145. But since widgets need the JupyterLab extension to be displayed in the notebook (and not all JupyterLite deployments may include it), we decided to leave the explicit call to `micropip` in the notebook.

**Reply by dsblank** [reply]
I see; thanks! 

It would be useful to have a `%pip` magic (just like IPython's) that would use micropip. Then jupyterlite's notebooks would be 100% the same with those other notebooks.

**Reply by bollwyvl** [reply]
Right. And adding a new package to `pyodide` can take _months_, because of maintainer load and deep coupling of patches to the compiler stack, cpython, etc. etc. 

We need that `conda-forge` `wasm-32` target _real bad_ to free up the bottleneck... nothing worse than a `conda`-_like_ build system that isn't actually...
